examples/common/chip-app-server/Server.cpp

```diff
@@ -61,16 +61,9 @@ class ServerCallback : public SecureSessionMgrDelegate
 
         ChipLogProgress(AppServer, "Packet received from %s: %zu bytes", src_addr, static_cast<size_t>(data_len));
 
-        HandleDataModelMessage(header, buffer, mgr);
-        buffer = NULL;
+        HandleDataModelMessage(header, std::move(buffer), mgr);
 
-    exit:
-        // HandleDataModelMessage calls Free on the buffer without an AddRef, if HandleDataModelMessage was not called, free the
-        // buffer.
-        if (buffer != NULL)
-        {
-            System::PacketBuffer::Free(buffer);
-        }
+    exit:;
```

Comment:
as we increasingly start having empty exit labels, I would propose to move to 'return' rather than 'gotol'. This will probably need to wait for #3880. Not for this PR, just thinking aloud (`exit:;` looks ugly).